### PR TITLE
Add randomised winner check

### PR DESCRIPTION
**Pull request by imsickofmaps**
For post M&L reward

### CODE REVIEW

**Comment by imsickofmaps** [comment]
@smn can I have a review please?

File: js_sandbox/lib/go-nike-ghr.js [hunk 1]
```diff
@@ -48,6 +48,16 @@ function GoNikeGHR() {
         }
     };
 
+    self.get_week_commencing = function(today) {
+        // today should be var today = new Date();
+        var date = today;
+        var day = date.getDay() || 7;
```
Comment (smn) [inline js_sandbox/lib/go-nike-ghr.js]: `date.getDay()` counts from zero, day `7` does not exist:

```
> d = new Date('Sunday Aug 25 2013 10:10:10 GMT+0200 (SAST)');
Sun Aug 25 2013 10:10:10 GMT+0200 (SAST)
> d.getDay();
0
> d = new Date('Saturday Aug 24 2013 10:10:10 GMT+0200 (SAST)');
Sat Aug 24 2013 10:10:10 GMT+0200 (SAST)
> d.getDay();
6
>
```

File: js_sandbox/lib/go-nike-ghr.js [hunk 2]
```diff
@@ -48,6 +48,16 @@ function GoNikeGHR() {
...
+        if (day !== 1)
+            date.setHours(-24 * (day - 1));
+        date.setHours(+2); // fixes crappy JS dates so we don't get Sunday
```
Comment (smn) [inline js_sandbox/lib/go-nike-ghr.js]: Doesn't this depend on the timezone of the host? Our servers are UTC not SAST.

File: js_sandbox/lib/go-nike-ghr.js [hunk 3]
```diff
@@ -287,16 +297,23 @@ function GoNikeGHR() {
                     }
                 }
                 var fields = null;
-
                 if (!question_id){
-                    // TODO: Add random airtime winner
+                    // Check for random airtime winner
+                    var winner = "false";
+                    if (im.config.airtime_reward_active){
+                        if (Math.floor((Math.random()*im.config.airtime_reward_chance)) === 0){
```
Comment (smn) [inline js_sandbox/lib/go-nike-ghr.js]: Please create a separate function that just returns a `true` or `false` so you can mock this function in a test and always return `true` or `false` and test for the the correct `ghr_airtime_winner` value written to the contact.

File: js_sandbox/lib/go-nike-ghr.js (hunk 2 above)
Comment (imsickofmaps) [inline js_sandbox/lib/go-nike-ghr.js]: Yeah - so it won't effect live but my dev laptop is on SAST. Bit of a hack but harmless?

File: js_sandbox/lib/go-nike-ghr.js (hunk 3 above)
Comment (imsickofmaps) [inline js_sandbox/lib/go-nike-ghr.js]: Do you have an example test for reliably validating a contact post-update? I tried to do this before but failed...

Comment (smn) [inline js_sandbox/lib/go-nike-ghr.js]: You can chain the test to the promise you get back from `tester.check_state({...})`.
I do something similar in [go-groups](https://github.com/smn/go-groups/blob/develop/test/test-go-groups.js#L235-L238)

File: js_sandbox/lib/go-nike-ghr.js (hunk 2 above)
Comment (smn) [inline js_sandbox/lib/go-nike-ghr.js]: But this isn't in a test?

**Comment by smn** [comment]
Did this go anywhere? Just trying to make sure it's not blocking on me.

**Pull request merged**